addons/sale_stock/tests/test_sale_stock.py

```diff
@@ -1050,23 +1050,36 @@ def test_packaging_propagation(self):
             'type': 'product',
         })
 
-        packaging = self.env['product.packaging'].create({
-            'name': 'box',
+        packOf10 = self.env['product.packaging'].create({
+            'name': 'PackOf10',
+            'product_id': product.id,
+            'qty': 10
+        })
+
+        packOf20 = self.env['product.packaging'].create({
+            'name': 'PackOf20',
             'product_id': product.id,
+            'qty': 20
         })
 
         so = self.env['sale.order'].create({
             'partner_id': self.partner_a.id,
             'order_line': [
                 (0, 0, {
                     'product_id': product.id,
-                    'product_uom_qty': 1.0,
+                    'product_uom_qty': 10.0,
                     'product_uom': product.uom_id.id,
-                    'product_packaging_id': packaging.id,
+                    'product_packaging_id': packOf10.id,
                 })],
         })
         so.action_confirm()
-        self.assertEqual(so.order_line.move_ids.product_packaging_id, packaging)
+        self.assertEqual(so.order_line.move_ids.product_packaging_id, packOf10)
+
+        so.order_line[0].write({'product_packaging_id': packOf20.id})
```

Comment:
Could you also increase the quantities ? And then check that the stock moves are merged ?